roles/bind-keycloak-apb/tasks/main.yml

```diff
@@ -24,23 +22,63 @@
   retries: 600
   delay: 2
 
--
-  name: "Create {{ service }}-bearer client in realm {{ namespace }}"
+- name: Get Keycloak Realms
+  uri: 
+    url: "{{ keycloak_route.stdout }}/auth/admin/realms/"
+    method: GET
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: 200
+    return_content: yes
+  register: keycloak_realms
+
+- set_fact: kc_realms={{ keycloak_realms.content | from_json | map(attribute='realm') | list}}
+
+- set_fact: realm={{ namespace }}
+  when: REALM_NAME is not defined
+
+- set_fact: realm={{ REALM_NAME }}
+  when: REALM_NAME is defined
+
+- name: Generate Keycloak Realm Template
+  template:
+    src: realm.json.j2
+    dest: /tmp/realm.json
+  when: "realm not in kc_realms"
+
+- name: "Create {{ realm }} realm"
+  uri:
+    url: "{{ keycloak_route.stdout }}/auth/admin/realms"
+    method: POST
+    body: "{{ lookup('file','/tmp/realm.json') }}"
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: 201
+  when: "realm not in kc_realms"
+
+- name: "Create {{ service }}-bearer client in {{ realm }} realm"
   uri:
-    url: "{{ keycloak_protocol }}://{{ keycloak_route.stdout }}/auth/admin/realms/{{ namespace }}/clients"
+    url: "{{ keycloak_route.stdout }}/auth/admin/realms/{{ realm }}/clients"
     method: POST
-    body: "{\"id\": \"{{ service}}-bearer\", \"secret\": \"{{ generated_password.stdout }}\",\"bearerOnly\":true}"
+    body: "{\"id\": \"{{ client_id }}\", \"clientId\": \"{{ service }}-bearer\", \"name\": \"{{ service }}\", \"secret\": \"{{ generated_secret.stdout }}\",\"bearerOnly\":true}"
     validate_certs: no
     body_format: json
     headers:
       Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
     status_code: 201, 409
+  register: create_client
```

Comment:
you can use failed when here:
```python
register: create_client
ailed_when: create_client.status == 409
```

Reply:
We actually would like to display a custom error message that would clearly state the error. I believe it will be used later on to add a last operation update.